mmengine/dist/dist.py

```diff
@@ -411,7 +412,10 @@ def _broadcast_object_list(object_list: List[Any],
     group_backend = get_backend(group)
     is_nccl_backend = group_backend == torch_dist.Backend.NCCL
     current_device = torch.device('cpu')
-    if is_nccl_backend:
+    if is_npu_available():
+        current_device = torch.npu.current_device()
+        object_sizes_tensor = object_sizes_tensor.to(current_device)
+    elif is_nccl_backend:
```

Comment:
Suggest making `get_backend` also support npu backend.

https://github.com/open-mmlab/mmengine/blob/13484aae9fbe66dd98f6a645267d9965239b7ce8/mmengine/dist/utils.py#L176

And then we can make this block consistent.

```python
    group_backend = get_backend(group)
    is_nccl_backend = group_backend == torch_dist.Backend
    is_hccl_backend = group_backend == 'hccl'
    if is_hccl_backend :
        current_device = torch.npu.current_device()
        object_sizes_tensor = object_sizes_tensor.to(current_device)
    elif is_nccl_backend:
```